src/runtime/hexagon/device/hexagon_stubapi.cc

```diff
@@ -0,0 +1,112 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * This Contribution is being provided by Qualcomm Technologies, Inc.,
+ * a Delaware corporation, or its subsidiary Qualcomm Innovation Center, Inc.,
+ * a California corporation, under certain additional terms and conditions
+ * pursuant to Section 5 of the Apache 2.0 license.  In this regard, with
+ * respect to this Contribution, the term "Work" in Section 1 of the
+ * Apache 2.0 license means only the specific subdirectory within the TVM repo
+ * (currently at https://github.com/dmlc/tvm) to which this Contribution is
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ * Qualcomm Technologies, Inc. and Qualcomm Innovation Center, Inc. retain
+ * copyright of their respective Contributions.
+ */
+#ifdef __ANDROID__
+#include "hexagon_stubapi.h"
+
+#include <dlfcn.h>
+#include <dmlc/logging.h>
+#include <stdint.h>
+#include <sys/stat.h>
+
+#include "hexagon_device_log.h"
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+
+StubAPI::StubAPI() {
+  struct stat sb;
+  if (!stat("/dev/subsys_cdsp", &sb)) {
+    enable_domains_ = true;
+    TVM_LOGD("CDSP subsystem present");
+  } else if (!stat("/dev/subsys_adsp", &sb)) {
+    enable_domains_ = false;
+    TVM_LOGD("ADSP subsystem present");
+  }
+
+  constexpr auto domain_lib_name = "libtvm_hexagon_remote_stub.so";
```

Comment:
How do we generate this .so? Is it contained in the `hexagon.cmake` build process?